pkg/storage/inmemory/inmemory.go

```diff
@@ -122,12 +127,20 @@ func (db *DB) List(_ context.Context, sinceVersion string) ([]*databroker.Record
 
 // Put replaces or inserts a record in the db.
 func (db *DB) Put(_ context.Context, id string, data *anypb.Any) error {
+	defer db.onchange.Broadcast()
 	db.replaceOrInsert(id, func(record *databroker.Record) {
 		record.Data = data
 	})
 	return nil
 }
 
+// Sync returns the underlying signal.Signal binding channel to the caller.
+// Then the caller can listen to the channel for detecting changes.
+func (db *DB) Sync(ctx context.Context) chan struct{} {
+	ch := db.onchange.Bind()
+	return ch
```

Comment:
We never unbind the channel... doesn't this result in a leak of channels? Every time a user calls Sync another one will be added and never removed.

Reply:
Oh sorry, the Unbind when ctx.Done() was removed accidently